trtools/prancSTR/tests/test_prancSTR.py

```diff
@@ -0,0 +1,375 @@
+import argparse
+import os
+
+import pytest
+
+from ..prancSTR import *
+
+# Set up base argparser
+@pytest.fixture
+def args(tmpdir):
+    args = argparse.ArgumentParser()
+    args.vcf = None
+    args.out = str(tmpdir / "test")
+    args.region = None
+    args.only_passing = False
+    args.debug = False
+    args.vcftype = "hipstr"
+    args.samples = None
+    args.quiet = True
+    args.output_all = False
+    args.readfield = "MALLREADS"
+    return args
+
+# Test no such file or directory
+def test_WrongFile(args, vcfdir):
+    fname = os.path.join(vcfdir, "test_non_existent.vcf")
+    if os.path.exists(fname):
+        os.remove(fname)
+    args.vcf = fname
+    retcode = main(args)
+    assert retcode==1
+
+    # real path but not VCF
+    fname = os.path.join(vcfdir, "CEU_test.vcf.gz.tbi")
+    args.vcf = fname
+    retcode = main(args)
+    assert retcode==1
+
+# Test bad output directory
+def test_BadOutdir(args, vcfdir, tmpdir):
+    fname = os.path.join(vcfdir, "test_hipstr.vcf")
+    args.vcf = fname
+    args.out = str(tmpdir / "bad/test")
+    retcode = main(args)
+    assert retcode==1
+
+    args.out = str(tmpdir)+os.sep
+    print(args.out)
+    retcode = main(args)
+    assert retcode==1
+
+# Test a good VCF file
+def test_RightFile(args, vcfdir):
+    fname = os.path.join(vcfdir, "test_hipstr.vcf")
+    args.vcftype = "auto"
+    args.vcf = fname
+    retcode = main(args)
+    assert retcode==0
+
+    args.quiet = False
+    args.debug = True
+    retcode = main(args)
+    assert retcode==0    
+
+    # Wrong VCF type
+    args.vcftype = "advntr"
+    retcode = main(args)
+    assert retcode==1
+
+# Test a multi-sample VCF file
+def test_MosaicCase(args, vcfdir):
+    fname = os.path.join(vcfdir, "CEU_test.vcf.gz")
+    args.vcf = fname
+    args.quiet = True
+    retcode = main(args)
+    assert retcode==0
+
+    args.quiet = False
+    retcode = main(args)
+    assert retcode==0
+
+    # Specific region
+    # Note: cyvcf2 handles region parsing and will
+    # output a warning if no intervals found
+    args.quiet = True
+    args.region = "chr1:987287-987288"
+    retcode = main(args)
+    assert retcode==0
+
+    # Samples list
+    args.samples = "NA12878"
+    retcode = main(args)
+    assert retcode==0
+
+    # Bad samples list should just give no output
+    args.samples = "XYZ"
+    retcode = main(args)
+    assert retcode==0
```

Comment:
It was intentional. Right now if you specify a sample that is not actually in the list, we just ignore it and continue. the help message for `--samples` warns about this. But if you think that is confusing we could add a check for that and abort mission if a sample is specified that is not in the VCF.

Reply:
I agree it's probably fine to just skip it and not abort. But perhaps it might be good to issue a warning just to let the user know how many samples were ignored in the end?

Reply:
I added a check to prancSTR.py that outputs a warning message if there is a sample missing